docs-site/content/0.19.0/guide/mongodb-full-text-search.md

```diff
@@ -0,0 +1,158 @@
+
+Hey there! 👋. This post gives you a guide to integrate Typesense server with MongoDB by setting up a trigger using MongoDB's Change Streams.
+
+![Typesense DynamoDB Integration Chart](@guide-images/typesense-mongodb/mongodb.svg)
+
+## Step 1: Install and Run Typesense
+
+To install and start typesense using docker issue the following docker command,
+
+```bash
+docker run -p 8108:8108 -v/tmp/typesense-data:/data typesense/typesense:0.19.0 \
+--data-dir /data --api-key=$TYPESENSE_API_KEY
+```
+
+Now, we can check if our Typsense server is ready to accept requests.
+
+```bash
+curl http://localhost:8108/health
+{"ok":true}
+```
+
+You can also run Typesense in other ways. Check out [Typesense Installation](https://typesense.org/docs/0.19.0/guide/install-typesense.html) for more details.
+
+## Step 2: Start a MongoDB Replica Set
+
+Here we will convert a standalone MongoDB instance to a replica set. 
+To convert to replica set:
+* Shutdown already running MongoDB server.
+* Start the MongoDB server by specifying -- replSet option
+```bash
+mongod --port "PORT" --dbpath "YOUR_DB_DATA_PATH" --replSet "REPLICA_SET_INSTANCE_NAME"
+```
+
+Check the status of replica set issuing the command `rs.status()` in mongo shell.
+
+## Step 3: Open a Change Stream
+
+We can open a change stream for MongoDB Replica Set from any of the data-bearing members.
+
+Example,
+```js
+const uri = '<MongoDB-URI>';
+const mongodbOptions = {
+    useNewUrlParser: true,
+    useUnifiedTopology: true
+}
+const client = new MongoClient(uri, mongodbOptions);
+await client.connect();
+const collection = client.db("sample").collection("books");
+const changeStream = collection.watch();
+changeStream.on('change', (next) => {
+    // process next document
+});
+```
+
+## Step 4: Create a Typesense Collection
+
+To use Typesense, we first need to create a client. Typesense supports multiple API clients including Javascript, Python, Ruby, PHP etc.
+
+To create the Javascript client, you need the API key of the Typesense server:
+
+```js
+import Typesense from 'typesense'
+
+let typesense = new Typesense.Client({
+  'nodes': [{
+    'host': 'localhost',
+    'port': '8108',
+    'protocol': 'http'
+  }],
+  'apiKey': '<API_KEY>',
+  'connectionTimeoutSeconds': 2
+})
+```
+Next, we will create a collection. A collection needs a schema, that represents how a douments would look like.
+
+```js
+let schema = {
+    'name': 'books',
+    'fields': [
+        { 'name': 'id', 'type': 'string', 'facet': false },
+        { 'name': 'name', 'type': 'string','facet': false },
+        { 'name': 'author', 'type': 'string', 'facet': false },
+        { 'name': 'year', 'type': 'int32', 'facet': true },
+    ],
+    'default_sorting_field': 'year',
+}
+await typesense.collections().create(schema);
+```
+
+## Step 5: Index documents to Typesense
+
+Next, we'll write functions to listen to change streams from MongoDB and write the changes to Typesense.
+
+A example MongoDB change streams response,
```

Comment:
A example -> Here's an example

Could you use colons instead of commas at the end sentences like this